ethereum/core/src/main/java/org/hyperledger/besu/ethereum/mainnet/headervalidationrules/BaseFeeMarketBlockHeaderGasPriceValidationRule.java

```diff
@@ -39,6 +40,11 @@ public BaseFeeMarketBlockHeaderGasPriceValidationRule(final BaseFeeMarket baseFe
   public boolean validate(final BlockHeader header, final BlockHeader parent) {
     try {
 
+      if (baseFeeMarket instanceof ZeroBaseFeeMarket) {
```

Comment:
Design wise this is too tight of coupling.  Instead there should be a method in `BaseFeeMarket` `isValidatesBasefee` that returns true or false based on a market config.

Or `isForkBlock` should be removed and replaced with a `validationMode(long blocknumber)` method that returns an enum with 3 states - `NONE`, `INITIAL`, and `ONGOING`.  Zero market would return `NONE`

Reply:
I like the validationMode idea, went with that 👍